components/post_view/channel_intro_message/messages/default/default.tsx

```diff
@@ -0,0 +1,122 @@
+import {t} from 'utils/i18n';
+import {ModalIdentifiers} from 'utils/constants';
+import ToggleModalButton from 'components/toggle_modal_button';
+import LocalizedIcon from 'components/localized_icon';
+import {Channel} from '@mattermost/types/channels';
+import {isArchivedChannel} from 'utils/channel_utils';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import AddMembersButton from '../../add_members_button';
+import BoardsButton from '../../boards_button';
+import SetHeaderButton from '../../set_header_button';
+
+type Props = {
+    channel: Channel;
+    usersLimit: number;
+    enableUserCreation?: boolean;
+    isReadOnly?: boolean;
+    teamIsGroupConstrained: boolean;
+}
+
+const messages = defineMessages({
+    readonlyDefault: {
+        id: t('intro_messages.readonly.default'),
+        defaultMessage: '**Welcome to {display_name}!**\\n \\nMessages can only be posted by system admins. Everyone automatically becomes a permanent member of this channel when they join the team.',
+    },
+    default: {
+        id: t('intro_messages.default'),
+        defaultMessage: '**Welcome to {display_name}!**\\n \\nPost messages here that you want everyone to see. Everyone automatically becomes a permanent member of this channel when they join the team.',
+    },
+    addGroupsToTeam: {
+        id: t('intro_messages.addGroupsToTeam'),
+        defaultMessage: 'Add other groups to this team',
+    },
+    beginning: {
+        id: t('intro_messages.beginning'),
+        defaultMessage: 'Beginning of {name}',
+    },
+
+});
+
+const DefaultIntroMessage = ({
+    channel,
+    usersLimit,
+    enableUserCreation,
+    isReadOnly,
+    teamIsGroupConstrained,
+}: Props) => {
+    let teamInviteLink = null;
+    const renderButtons = !isReadOnly && !isArchivedChannel(channel);
+
+    if (!isReadOnly && enableUserCreation) {
+        teamInviteLink = (
+            <TeamPermissionGate
+                teamId={channel.team_id}
+                permissions={[Permissions.INVITE_USER]}
+            >
+                <TeamPermissionGate
+                    teamId={channel.team_id}
+                    permissions={[Permissions.ADD_USER_TO_TEAM]}
+                >
+                    {teamIsGroupConstrained ? (
+                        <ToggleModalButton
+                            className='intro-links color--link'
+                            modalId={ModalIdentifiers.ADD_GROUPS_TO_TEAM}
+                            dialogType={AddGroupsToTeamModal}
+                            dialogProps={{channel}}
+                        >
+                            {/* MM-46602: convert to compass icon after localization is added */}
+                            <LocalizedIcon
+                                className='fa fa-user-plus'
+                                title={{id: t('generic_icons.add'), defaultMessage: 'Add Icon'}}
+                            />
```

Comment:
localization is now live in the latest compass-icon version ... feel free to replace that

Reply:
Is this a request to replace `LocalizedIcon`?  I'm not sure how this one should be handled/changed?

Reply:
well, you can just use the icon comonents from compass-icons package and set the `title` and `aria-label` prop for them. This is basically the same the `LocalizedIcon` does, but it still renders the icon font.

So to give you an example you can just use this:

```
<AccountPlusOutlineIcon
    size={18}
    title={formatMessage({id: 'generic_icons.add', defaultMessage: 'Add Icon'})}
    aria-label={'<ADD_ARIA_STRING_HERE >'}
/>
```

with the latest version of compass-icons we included 'currentColor' as the default value for the `color` prop, so when the icon should have the same color as the text (if present) you can just set a color property on the parent SCSS class and you are good to go

Reply:
We don't have a non-outlined compass-icon for AccountPlusIcon.  I've created a ticket for adding it later and created a ticket https://mattermost.atlassian.net/browse/MM-47138

Reply:
Here's the outlined icon next to the current icon which is filled. <img width="288" alt="image" src="https://user-images.githubusercontent.com/7575921/191068996-7ffbf518-0eb2-498f-b7d1-8cd8ce065cdf.png">

Reply:
@matthewbirtch are we good with the outlined version as well?

Reply:
It should be using the outlined version